src/number_theory/fermat_psp.lean

```diff
@@ -0,0 +1,522 @@
+/-
+Copyright (c) 2022 Niels Voss. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Niels Voss
+-/
+import data.nat.prime
+import field_theory.finite.basic
+
+/-!
+# Fermat Pseudoprimes
+
+In this file we define Fermat pseudoprimes: composite numbers that pass the Fermat primality test.
+A natural number `n` passes the Fermat primality test to base `b` (and is therefore deemed a
+"probable prime") if `n` divides `b^(n - 1) - 1`. `n` is a Fermat pseudoprime to base `b` if `n` is
+a composite number that passes the Fermat primality test to base `b` and is coprime with `b`.
+
+Fermat pseudoprimes can also be seen as composite numbers for which Fermat's little theorem holds
+true.
+
+## Main Results
+
+The main definitions for this file are
+
+- `fermat_psp.probable_prime`: A number `n` is a probable prime to base `b` if it passes the Fermat
+  primality test; that is, `b` divides `b^(n - 1) - 1`
+- `fermat_psp`: A number `n` is a pseudoprime to base `b` if it is a probable prime to base `b`, is
```

Comment:
Do I understand https://en.wikipedia.org/wiki/Carmichael_number#Overview correctly that these numbers are also called Carmichael numbers? If so, please mention that here and in the docstring.

Reply:
Good point. Maybe you can still indicate the relation between the two notions in the module docstring at the top of the file?

Reply:
I added a short note about this at the top of the file.